ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/BackupReplayFile.java

```diff
@@ -79,16 +89,39 @@ public String getPath() {
     return file.getAbsolutePath();
   }
 
-  public void write(final ConsumerRecord<byte[], byte[]> record) throws IOException {
-    if (writer == null) {
-      throw new IOException("Write permission denied.");
-    }
-
+  private static void appendRecordToFile(
+      final ConsumerRecord<byte[], byte[]> record,
+      final File file,
+      final Filesystem filesystem
+  ) throws IOException {
+    final FileOutputStream writer = createWriter(file, filesystem);
     writer.write(record.key());
     writer.write(KEY_VALUE_SEPARATOR_BYTES);
     writer.write(record.value());
     writer.write(NEW_LINE_SEPARATOR_BYTES);
-    writer.flush();
+    writer.close();
+  }
+
+  public void write(final ConsumerRecord<byte[], byte[]> record) throws IOException {
+    if (!writable) {
+      throw new IOException("Write permission denied.");
+    }
+    final File dirty = dirty(file);
+    final File tmp = tmp(file);
```

Comment:
is the `tmp` file what you call the `live` file in the PR description? or is `tmp` just temporary file and the `file` we pass in is the `live` file

Reply:
Not sure what you mean by the file we pass in. The live file is at the path stored in `file` - it's the file we're trying to update.